### PR TITLE
Introduction of batch-2.1 

### PR DESCRIPTION
**Pull request by cgianfra**
Jared: I think there's still a piece missing from the changes to accommodate Record moving to DataRecord. This abstraction we came up with builds fine, and I don't have those errors in the FATs I've been running anymore, but I haven't gotten to manual testing with an actual environment that would fully expose the error. I believe I still need to make a new custom feature or two to load the right pieces in the right context.

### CODE REVIEW

File: dev/com.ibm.jbatch.container/.classpath [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <classpath>
 	<classpathentry kind="src" output="bin" path="src"/>
 	<classpathentry kind="src" output="bin_test" path="test"/>
-	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+	<classpathentry exported="true" kind="con" path="aQute.bnd.classpath.container"/>
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.container/.classpath]: Odd change.  Any reason?

File: dev/com.ibm.jbatch.internal/bnd.bnd [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#######################################################################
+Export-Package:\
+ com.ibm.jbatch.container.persistence.jpa;provide:=true;version=1.0.0
+
+# Using version=! in order to not have a version attached to the import for packages that were removed
+# from Java after Java 8.  Doing this keeps the import like before Java 11 support. It will get the 
+# packages from Java when using Java 8 or earlier and from the new shipped bundles for Java 9 and later.
+##
+# Using '*' below causes problems with our two separate src dirs
+##
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal/bnd.bnd]: Not doing this so delete the comment please.
```suggestion
```

File: dev/com.ibm.jbatch.internal/build.properties [hunk 3]
```diff
@@ -0,0 +1,13 @@
+###############################################################################
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal/build.properties]: What is the point of this file?

File: dev/com.ibm.jbatch.internal/.project [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>com.ibm.jbatch.internal</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal/.project]: This should have the bnd builder as well.

File: dev/com.ibm.jbatch.internal/.classpath [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal/.classpath]: This should be specifically Java 8 and not just JRE_CONTAINER.
Also the bnd.classpath.container should be in here as well.

File: dev/com.ibm.jbatch.container/transformed10.bnd [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/transform10.props
+
+Bundle-Name: Java Batch RI core runtime implementation Jakarta
+Bundle-SymbolicName: com.ibm.jbatch.container.jakarta.ee10
+Bundle-Description: Java Batch RI core runtime implementation; Jakarta Enabled
+
+#-includeresource: \
+#  @${repo;io.openliberty.jbatch.internal.jakarta.ee10}!/com/ibm/jbatch/container/persistence/jpa/*.class
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.container/transformed10.bnd]: Obviously the projects are missing that you are referencing in the bnd files in this project.  You will need to create them.

**Review by jhanders34 — CHANGES_REQUESTED** [review]

File: dev/com.ibm.jbatch.container/.classpath (hunk 1 above)
Comment (cgianfra) [reply dev/com.ibm.jbatch.container/.classpath]: I don't remember making this myself, maybe came from doing something in Eclipse? I'll remove it and make sure it doesn't break anything.

File: dev/com.ibm.jbatch.internal/build.properties (hunk 3 above)
Comment (cgianfra) [reply dev/com.ibm.jbatch.internal/build.properties]: Carried over the build config files from jbatch.container when I made the project, but it seems this isn't needed so I'll remove it.

File: dev/com.ibm.jbatch.internal.ee10/.classpath [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="src"/>
+        <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+        <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal.ee10/.classpath]:
```suggestion
        <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11"/>
```

File: dev/com.ibm.jbatch.internal/bnd.bnd [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+
+-sub: *.bnd
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal/bnd.bnd]: No reason to do a sub bundle if there is only 1 bundle.  Just put what is in the original.bnd into this bnd file.

File: dev/com.ibm.jbatch.internal.ee10/bnd.bnd [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+
+-sub: *.bnd
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal.ee10/bnd.bnd]: No reason to do a sub bundle if there is only 1 bundle.  Just put what is in the original.bnd into this bnd file.

File: dev/com.ibm.jbatch.internal.ee10/bnd.bnd [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+bVersion=1.0
+
+
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal.ee10/bnd.bnd:23]:
```suggestion
javac.source: 11
javac.target: 11

publish.wlp.jar.disabled: true
```

File: dev/com.ibm.jbatch.internal/bnd.bnd [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+bVersion=1.0
+
+
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal/bnd.bnd]:
```suggestion
publish.wlp.jar.disabled: true
```

File: dev/com.ibm.jbatch.container/bnd.bnd [hunk 12]
```diff
@@ -115,12 +115,14 @@ instrument.classesExcludes: com/ibm/jbatch/container/internal/resources/*.class
 	com.ibm.websphere.appserver.spi.threading;version=latest,\
 	com.ibm.websphere.javaee.cdi.1.0;version=latest,\
 	com.ibm.websphere.appserver.api.kernel.service;version=latest,\
-	com.ibm.websphere.appserver.thirdparty.eclipselink;version=latest, \
+	com.ibm.websphere.appserver.thirdparty.eclipselink;version=latest,\
+	io.openliberty.persistence.3.1.thirdparty;version=latest,\
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.container/bnd.bnd]: don't know if you need io.openliberty.persistence.3.1.thirdparty in the buildpath now.

**Review by jhanders34 — CHANGES_REQUESTED** [review]
The new com.ibm.jbatch.internal* projects are missing their .settings directories.  Should have at least bndtools.core.prefs and org.eclipse.jdt.core.prefs.

File: dev/com.ibm.jbatch.internal.ee10/.settings/org.eclipse.jdt.core.prefs [hunk 13]
```diff
@@ -0,0 +1,286 @@
...
+org.eclipse.jdt.core.formatter.comment.indent_root_tags=true
+org.eclipse.jdt.core.formatter.number_of_blank_lines_at_beginning_of_method_body=0
+org.eclipse.jdt.core.codeComplete.localPrefixes=
+org.eclipse.jdt.core.formatter.insert_space_before_opening_paren_in_synchronized=insert
+org.eclipse.jdt.core.formatter.insert_space_before_closing_paren_in_switch=do not insert
+org.eclipse.jdt.core.formatter.insert_space_after_opening_paren_in_method_invocation=do not insert
+org.eclipse.jdt.core.formatter.insert_space_after_comma_in_for_inits=insert
+org.eclipse.jdt.core.formatter.insert_space_before_opening_brace_in_annotation_type_declaration=insert
+org.eclipse.jdt.core.formatter.insert_space_after_opening_brace_in_array_initializer=insert
+org.eclipse.jdt.core.compiler.compliance=1.8
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal.ee10/.settings/org.eclipse.jdt.core.prefs]:
```suggestion
org.eclipse.jdt.core.compiler.compliance=11
```

File: dev/com.ibm.jbatch.internal.ee10/.settings/org.eclipse.jdt.core.prefs [hunk 14]
```diff
@@ -0,0 +1,286 @@
...
+org.eclipse.jdt.core.formatter.comment.format_html=false
+org.eclipse.jdt.core.formatter.insert_space_before_comma_in_explicitconstructorcall_arguments=do not insert
+org.eclipse.jdt.core.formatter.insert_space_before_closing_angle_bracket_in_parameterized_type_reference=do not insert
+org.eclipse.jdt.core.formatter.insert_new_line_after_label=do not insert
+org.eclipse.jdt.core.formatter.insert_space_before_closing_paren_in_annotation=do not insert
+org.eclipse.jdt.core.codeComplete.argumentSuffixes=
+org.eclipse.jdt.core.formatter.number_of_empty_lines_to_preserve=1
+org.eclipse.jdt.core.formatter.insert_space_after_comma_in_enum_declarations=insert
+org.eclipse.jdt.core.formatter.comment.insert_new_line_before_root_tags=insert
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.8
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal.ee10/.settings/org.eclipse.jdt.core.prefs]:
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=11
```

File: dev/com.ibm.jbatch.internal.ee10/.settings/org.eclipse.jdt.core.prefs [hunk 15]
```diff
@@ -0,0 +1,286 @@
...
+org.eclipse.jdt.core.formatter.insert_space_after_opening_paren_in_while=do not insert
+org.eclipse.jdt.core.formatter.insert_space_before_comma_in_method_declaration_throws=do not insert
+org.eclipse.jdt.core.formatter.insert_space_before_comma_in_for_increments=do not insert
+org.eclipse.jdt.core.formatter.insert_space_after_opening_paren_in_method_declaration=do not insert
+org.eclipse.jdt.core.formatter.insert_space_before_closing_paren_in_constructor_declaration=do not insert
+org.eclipse.jdt.core.formatter.insert_space_before_opening_paren_in_switch=insert
+org.eclipse.jdt.core.formatter.insert_space_before_comma_in_array_initializer=do not insert
+org.eclipse.jdt.core.formatter.insert_space_after_colon_in_for=insert
+org.eclipse.jdt.core.formatter.insert_space_after_opening_paren_in_annotation=do not insert
+org.eclipse.jdt.core.compiler.source=1.8
```
Comment (jhanders34) [inline dev/com.ibm.jbatch.internal.ee10/.settings/org.eclipse.jdt.core.prefs]:
```suggestion
org.eclipse.jdt.core.compiler.source=11
```

**Review by jhanders34 — APPROVED** [review]

**Comment by cgianfra** [comment]
#build

**Comment by cgianfra** [comment]
#build

**Comment by cgianfra** [comment]
#libbybot

**Pull request merged**